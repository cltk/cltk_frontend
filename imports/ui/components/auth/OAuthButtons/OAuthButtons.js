import React from 'react';

const OAuthButtons = ({ handleFacebook, handleGoogle, handleTwitter }) => (
	<div className="at-oauth">
		{handleFacebook &&
			<button
				className="btn at-social-btn"
				id="at-facebook"
				name="facebook"
				onClick={handleFacebook}
			>
				<i className="fa fa-facebook" /> Sign In with Facebook
			</button>}

		{handleGoogle &&
			<button
				className="btn at-social-btn"
				id="at-google"
				name="google"
				onClick={handleGoogle}
			>
				<i className="fa fa-google" /> Sign In with Google
			</button>}

		{handleTwitter &&
			<button
				className="btn at-social-btn"
				id="at-twitter"
				name="twitter"
				onClick={handleTwitter}
			>
				<i className="fa fa-twitter" /> Sign In with Twitter
			</button>}
	</div>
);
OAuthButtons.propTypes = {
	handleFacebook: React.PropTypes.func,
	handleGoogle: React.PropTypes.func,
	handleTwitter: React.PropTypes.func,
};
OAuthButtons.defaultProps = {
	handleFacebook: null,
	handleGoogle: null,
	handleTwitter: null,
};

export default OAuthButtons;
