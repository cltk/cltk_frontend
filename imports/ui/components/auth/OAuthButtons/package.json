{
  "name": "OAuthButtons",
  "version": "0.0.1",
  "private": true,
  "main": "./OAuthButtons.js"
}
