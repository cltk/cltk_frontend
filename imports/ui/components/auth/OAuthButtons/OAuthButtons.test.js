import React from 'react';
import renderer from 'react-test-renderer';

// component:
import OAuthButtons from './OAuthButtons';

describe('OAuthButtons', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<OAuthButtons />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
