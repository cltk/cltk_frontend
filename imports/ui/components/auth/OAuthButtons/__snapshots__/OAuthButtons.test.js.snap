// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OAuthButtons renders correctly 1`] = `
<div
  className="at-oauth"
/>
`;
